File: choremaster-backend/src/main/java/de/philgenstock/choremaster/user/UserController.java
package de.philgenstock.choremaster.user;

import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@AllArgsConstructor
public class UserController {

    private final UserApplicationService userApplicationService;

    @PostMapping("/login")
    public void login(@RequestBody UserDto userDto) {
        userApplicationService.login(userDto);
    }
}

File: choremaster-rest/rest/src/main/kotlin/de/philgenstock/choremaster/user/UserController.kt
package de.philgenstock.choremaster.user

import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/user")
class UserController(
    private val userService: UserService,
) {
    @PostMapping("/login")
    fun login(
        @RequestParam("token") googleToken: String,
    ) = userService.login(googleToken)
}
